# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/gzip/gzip-1.3.3-addsuffix.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./zmore.in.addsuffix	2004-12-13 09:55:48.000000000 +0100
+++ ./zmore.in	2004-12-13 09:58:24.641766288 +0100
@@ -66,7 +66,7 @@
 	--v*) exec echo "$version";;
 	esac
 
-	< "$FILE" || continue
+	if test -e $FILE || test -e $FILE.gz; then
 	if test $FIRST -eq 0; then
 		echo $n1 "--More--(Next file: $FILE)$n2"
 		stty $cb -echo 2>/dev/null
@@ -84,5 +84,7 @@
 	if test -t 1; then
 		FIRST=0
 	fi
+	else < $FILE
+	fi
     done
 fi
